File: Modules/RoxieMobile.NetworkingApi/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/config/HttpClientConfig.java
package com.roxiemobile.networkingapi.network.rest.config;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

import javax.net.ssl.HostnameVerifier;
import javax.net.ssl.SSLSocketFactory;
import javax.net.ssl.X509TrustManager;

import okhttp3.CertificatePinner;
import okhttp3.Interceptor;

public interface HttpClientConfig {

// MARK: - Methods

    /**
     * TODO
     */
    int connectTimeout();

    /**
     * TODO
     */
    int readTimeout();

    /**
     * TODO
     */
    @NotNull List<Interceptor> interceptors();

    /**
     * TODO
     */
    @NotNull List<Interceptor> networkInterceptors();

    /**
     * TODO
     */
    default @Nullable CertificatePinner certificatePinner() {
        return null;
    }

    /**
     * TODO
     */
    default @Nullable HostnameVerifier hostnameVerifier() {
        return null;
    }

    /**
     * TODO
     */
    default @Nullable SSLSocketFactory sslSocketFactory() {
        return null;
    }

    /**
     * TODO
     */
    default @Nullable X509TrustManager trustManager() {
        return null;
    }
}

File: Modules/RoxieMobile.NetworkingApi/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/config/HttpClientConfig.kt
package com.roxiemobile.networkingapi.network.rest.config

import com.roxiemobile.networkingapi.network.ssl.TlsConfig
import okhttp3.Interceptor

interface HttpClientConfig: Cloneable {

// MARK: - Properties

    /**
     * TODO
     */
    val connectionTimeout: Long

    /**
     * TODO
     */
    val readTimeout: Long

    /**
     * TODO
     */
    val tlsConfig: TlsConfig?

    /**
     * TODO
     */
    val interceptors: List<Interceptor>?

    /**
     * TODO
     */
    val networkInterceptors: List<Interceptor>?

// MARK: - Methods

    /**
     * TODO
     */
    public override fun clone(): HttpClientConfig
}
